sql/core/src/main/scala/org/apache/spark/status/api/v1/sql/SqlResource.scala

```diff
@@ -94,7 +94,8 @@ private[v1] class SqlResource extends BaseAppResource {
 
     val duration = exec.completionTime.getOrElse(new Date()).getTime - exec.submissionTime
     val planDetails = if (planDescription) exec.physicalPlanDescription else ""
-    val nodes = if (details) printableMetrics(graph.allNodes, exec.metricValues) else Seq.empty
+    val nodes = if (details) printableMetrics(graph.allNodes,
+      Option(exec.metricValues).getOrElse(Map())) else Seq.empty
```

Comment:
```suggestion
      Option(exec.metricValues).getOrElse(Map.empty)) else Seq.empty
```

Reply:
nit: I wouldn't use `Option` here. Just check for `null`, e.g. `val metricValues = if (exec.metricValues != null) exec.metricValues else Map.empty`
But I guess the extra memory allocation for `Option` is not a big deal here